[MIOpen] Convert embed_sqlite ctest to GTest

kokolchin opened the pull request:
## Motivation

This PR converts the `embed_sqlite` test from the legacy ctest framework to Google Test format, aligning with MIOpen's migration strategy to standardize on Google Test for all unit tests. The test verifies that embedded SQLite databases (PerfDb and FindDb) function correctly when MIOpen is built with `MIOPEN_EMBED_DB` enabled, ensuring that database embedding works as expected for embedded system deployments.

## Technical Details

* Converted `embed_sqlite.cpp` from ctest to Google Test format
* The test follows MIOpen gtest naming conventions using `GPU_EmbedSQLite_FP32` test suite name
* Uses `TEST_F` (fixture-based test) structure with `::testing::Test` base class
* Removed dependency on custom `test_driver` framework
* Tests verify embedded database functionality:
  - **PerfDb verification**: Checks that embedded PerfDb files can be found in `miopen_data()` using `make_object_file_name()`, and verifies that `GetDb()` can successfully find records for convolution problems
  - **FindDb verification**: Validates that `FindDbRecord` works correctly with embedded databases and returns non-empty results
  - Tests use a convolution problem descriptor (128x1024x14x14 input, 2048x1024x1x1 filter) to verify database lookups
* Test is conditionally compiled with `#if MIOPEN_EMBED_DB` to only build when embedding is enabled

Related: https://github.com/ROCm/MIOpen/wiki/GTest-development#naming

## Test Plan

* Verified test naming convention compliance using `gtest_formating_checks.py`
* Test compiles successfully with `MIOPEN_EMBED_DB` enabled
* Test execution to be verified via GitHub Actions CI
* Verified that the old ctest file (`embed_sqlite.cpp`) has been removed

## Test Result

* ✅ Naming convention check passed
* ✅ Compilation successful
* ⏳ CI tests pending (execution verification)

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

kokolchin committed:
Fix compilation errors in check_numerics_test: use reference for Handle, add includes

Convert embed_sqlite from ctest to Google Test

Format embed_sqlite.cpp.

Fix embed_sqlite test: ensure test is always visible to gtest even when MIOPEN_EMBED_DB is not defined

Format check_numerics_test.cpp.

Add missing newline at end of check_numerics_test.cpp

Fix gtest naming convention: rename Test to EmbedSQLite

Change test name from generic 'Test' to descriptive 'EmbedSQLite'
to comply with gtest naming conventions

Restore ctest check_numerics_test.cpp and remove gtest version

Remove test/gtest/check_numerics_test.cpp (gtest version is under development in feature/gtest-check-numerics)
Restore test/check_numerics_test.cpp (ctest version) from develop

Move TEST macros outside the anonymous namespace.

Use SPDX licens format. Move the open brace of the anonymous namespace

Remove the anonymous namespace

Remove accidentally included check_numerics_test.cpp (not related to embed_sqlite)

Update embed_sqlite test to follow GTest development guidelines: use INSTANTIATE_TEST_SUITE_P, anonymous namespace, and SetUp() override

Format using make format

Use ValuesIn() instead of Values() in embed_sqlite gtest

Move IsTestSupportedForDevice() inside #if MIOPEN_EMBED_DB and add explanatory comment

Remove unnecessary comment

Fix compilation errors in embed_sqlite test

- Change IsTestSupportedForDevice to IsTestSupportedByDevice
- Change parameter from miopen::Handle to Gpu::All
- Remove unused local IsTestSupportedForDevice function
- Add gtest_common.hpp include

Refactor embed_sqlite test structure

- Update test to utilize INSTANTIATE_TEST_SUITE_P for parameterized tests
- Implement SetUp() override for better test organization
- Ensure compliance with GTest development guidelines

Remove unrelated cba_inference and cbna_inference files from embed_sqlite branch

Restore old ctest files from develop

fix: remove duplicate test definitions in embed_sqlite.cpp

Address reviewer feedback: use miopen::fs::path in-place and remove namespace alias

Reset unrelated cbna_inference.cpp to match develop

Format

Remove redundant IsTestSupportedByDevice(Gpu::All) check and apply clang-format

Align embed_sqlite.cpp with GTest guidelines: add anonymous namespace and ensure PRNG reset

Rename GPU_EmbedSQLite_FP32 to CPU_EmbedSQLite_NONE as it only performs database lookups without GPU kernels

vasily-vm commented on projects/miopen/test/gtest/embed_sqlite.cpp:
```diff
@@ -0,0 +1,88 @@
+#include <miopen/find_db.hpp>
+#include <miopen/filesystem.hpp>
+
+namespace fs = miopen::fs;
```
remove and use in-place `miopen::fs::...`

vasily-vm reviewed (commented):
overall looks good

--please check, does old CTest was removed
checked: it was removed by mistake in [PR2838](https://github.com/ROCm/rocm-libraries/pull/2838)

kokolchin commented:
GTest takes 26 ms.

Vsevolod1983 commented:
It looks like there is no any GPU calculations
Might it be better to rename the test from GPU_EmbedSQLite_FP32  to CPU_EmbedSQLite_NONE ?

Vsevolod1983 commented:
Also, let's ping @randyspauldingamd 
This test was removed in your PR in the past. Did you intentionally remove it or by mistake?

kokolchin commented:
> It looks like there is no any GPU calculations Might it be better to rename the test from GPU_EmbedSQLite_FP32 to CPU_EmbedSQLite_NONE ?

Renamed.

kokolchin committed:
Fix solver gtest kernel comparison type.

Use std::string for expected kernel in ConstructTest to ensure content comparison instead of C-string pointer comparison in EXPECT_EQ.

Vsevolod1983 commented:
Please remove unnecessary changes

geomin12 reviewed (commented):
from TheRock CI, looks good - however, we should leave this review specifically for MIOpen folks to make sure it is right

kokolchin committed:
Fix solver test expectations for no-search DB behavior.

Adjust expected kernel names for searchable test cases to SearchableTestSolver.NoSearch when records are reused from DB without search.

kokolchin committed:
Sync solver gtest with develop (PR 4745).

Replace test/gtest/solver.cpp with the current develop version to align expected searchable-solver behavior and remove branch-specific divergence.
